src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -6180,7 +6421,8 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_StringConstant(const ASR::StringConstant_t &x) {
-        tmp = builder->CreateGlobalStringPtr(x.m_s);
+        std::string s = unescape_string(al, x.m_s);
```

Comment:
unescape_string is handled in the frontend itself: https://github.com/lcompilers/lpython/pull/1762/files#diff-48a7161397f9eb8b730f0cbec2f11ce1471aac961dc6b16dc558cf161c375054